java-client/session-dagger/src/test/java/io/deephaven/client/MultiClientSharedIdTest.java

```diff
@@ -54,4 +60,54 @@ public void testHandoff() throws TableHandle.TableHandleException, InterruptedEx
         // Ensure that we can resolve from query scope.
         client2.execute(destId.ticketId().table());
     }
+
+    @Test
+    public void testFailsPublishingFromQueryScope() throws Exception {
+        final TimeTable sourceTable = TimeTable.of(Duration.ofSeconds(1));
+        final TableHandle remoteSource = session.execute(sourceTable);
+        final ScopeId scopeId = new ScopeId("test_time_table");
+        final SharedId sharedId = SharedId.newRandom();
+
+        // Let's publish the source table to the query scope.
+        session.publish(scopeId, remoteSource).get();
+
+        // This publish should fail because the source is not a session owned export.
+        try {
+            session.publish(sharedId, scopeId).get();
+            Assert.statementNeverExecuted();
+        } catch (final ExecutionException err) {
+            // This is expected.
+            Assert.eqTrue(err.getCause() instanceof StatusRuntimeException,
+                    "err.getCause() instanceof StatusRuntimeException");
+        }
+    }
+
+    @Test
+    public void testSharePublishCompletesImmediately() throws Exception {
+        final TicketId ticketId = new TicketId(ExportTicketHelper.exportIdToBytes(1));
+        final SharedId sharedId = SharedId.newRandom();
+
+        // Let's publish the source table to the shared id.
+        session.publish(sharedId, ticketId).get();
```

Comment:
We should probably follow up here with an export against sharedId to make sure it succeeds?

Then, does it make sense to test now what happens if ticketId fails? I know we were talking about an exception "firewall" as errors propgate to across the sharedId boundary.